data-prepper-plugins/aggregate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/aggregate/actions/CountAggregateAction.java

```diff
@@ -0,0 +1,100 @@
+import org.opensearch.dataprepper.plugins.processor.aggregate.AggregateActionInput;
+import org.opensearch.dataprepper.plugins.processor.aggregate.AggregateActionResponse;
+import org.opensearch.dataprepper.plugins.processor.aggregate.GroupState;
+import io.opentelemetry.proto.metrics.v1.AggregationTemporality;
+
+import java.time.Instant;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Optional;
+
+/**
+ * An AggregateAction that combines multiple Events into a single Event. This action will add the unique keys of each smaller Event to the overall groupState,
+ * and will create a combined Event from the groupState on concludeGroup. If smaller Events have the same keys, then these keys will be overwritten with the keys of the
+ * most recently handled Event.
+ * @since 1.3
+ */
+@DataPrepperPlugin(name = "count", pluginType = AggregateAction.class, pluginConfigurationType = CountAggregateActionConfig.class)
+public class CountAggregateAction implements AggregateAction {
+    static final String EVENT_TYPE = "event";
+    static final String SUM_METRIC_NAME = "count";
+    static final String SUM_METRIC_DESCRIPTION = "Number of events";
+    static final String SUM_METRIC_UNIT = "1";
+    static final boolean SUM_METRIC_IS_MONOTONIC = true;
+    public final String countKey;
+    public final String outputFormat;
+    private long startTimeNanos;
+
+    @DataPrepperPluginConstructor
+    public CountAggregateAction(final CountAggregateActionConfig countAggregateActionConfig) {
+        this.countKey = countAggregateActionConfig.getCountKey();
+        this.outputFormat = countAggregateActionConfig.getOutputFormat();
+    }
+
+    private long getCurrentTimeNanos() {
+        final long NANO_MULTIPLIER = 1_000 * 1_000 * 1_000;
+        Instant now = Instant.now();
+        long currentTimeNanos = now.getEpochSecond() * NANO_MULTIPLIER + now.getNano();
+        return currentTimeNanos;
+    }
+
+    @Override
+    public AggregateActionResponse handleEvent(final Event event, final AggregateActionInput aggregateActionInput) {
+        final GroupState groupState = aggregateActionInput.getGroupState();
+        if (groupState.get(countKey) == null) {
+            startTimeNanos = getCurrentTimeNanos();
+            groupState.putAll(aggregateActionInput.getIdentificationKeysMap());
+            groupState.put(countKey, 1);
+        } else {
+            Integer v = (Integer)groupState.get(countKey) + 1;
+            groupState.put(countKey, v);
+        } 
+        return AggregateActionResponse.nullEventResponse();
+    }
+
+    @Override
+    public Optional<Event> concludeGroup(final AggregateActionInput aggregateActionInput) {
+        GroupState groupState = aggregateActionInput.getGroupState();
+        Event event;
+        if (outputFormat == OutputFormat.DEFAULT.toString()) {
+            event = JacksonEvent.builder()
+                .withEventType(EVENT_TYPE)
+                .withData(groupState)
```

Comment:
I think the time here should be named `startTime` and be output as an ISO-8601 string. You might need to refactor how you handle the time to make this work. I don't think you need to convert to nano. Just store an `Instant` in the group state.

Reply:
The SUM Metric in the DataPrepper Metric model has both "time" and "startTime". Not sure if it was mistake.

Reply:
Existing code in .`/data-prepper-plugins/otel-metrics-raw-processor/` uses  the API `OTelProtoCodec.convertUnixNanosToISO8601(nanos)` to get ISO-8601 format. I am using the same here.

Reply:
No. The current implementation of "default" (to be renamed to "raw") does not have time field in it. It just has "count". I am suggesting that for "raw" output, we should put start time and/or end time at aggregate processor level (not at count action level) based on "add_keys" config option